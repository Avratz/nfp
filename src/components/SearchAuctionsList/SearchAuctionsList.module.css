.auctionCards {
  display: flex;
  flex-direction: column;
  gap: 2rem;
  padding: 0.5rem 0;
  width: 100%;
}

.auctionCardsContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
}

.noMatch {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1rem;
  width: 100%;
}

.loading {
  display: flex;
  align-items: center;
  gap: 0.25rem;
  width: 100%;
}

@media screen and (min-width: 768px) {
  .auctionCards {
    flex-direction: row;
    flex-wrap: wrap;
    padding: 0;
  }
  .auctionCardsContainer {
    width: 60%;
  }
}

@media screen and (min-width: 1200px) {
  .auctionCards {
    justify-content: flex-start;
  }
  .auctionCardsContainer {
    width: 80%;
  }
}
