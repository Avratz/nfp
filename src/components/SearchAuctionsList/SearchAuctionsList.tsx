import Image from 'next/image'
import { useDeferredValue } from 'react'

import { useSearchAuctions } from '@/hooks/useSearchAuctionsContext'

import { Button } from '@/ui/Button/Button'
import { Heading } from '@/ui/Heading/Heading'
import { Icon } from '@/ui/Icon/Icon'

import { AuctionCard } from '@/components/AuctionCard/AuctionCard'

import styles from './SearchAuctionsList.module.css'

export function SearchAuctionsList() {
  const { auctions } = useSearchAuctions()
  const deferredAuctions = useDeferredValue(auctions)
  return (
    <div className={styles.auctionCardsContainer}>
      <div className={styles.auctionCards}>
        {deferredAuctions.length === 0 ? (
          <div className={styles.noMatch}>
            <Image src="/images/error.png" alt="" width={215} height={160} />
            <Heading as="h3" size="2xl">
              Im terrible sorry about all this.
            </Heading>
            <Heading as="h4">Please try again with another filter.</Heading>
          </div>
        ) : (
          deferredAuctions.map((auction, index) => (
            <AuctionCard
              key={index}
              image={auction.media.image}
              author={auction.author}
              instantPrice={auction.instantPrice}
              stock={auction.stock}
              highestBid={auction.highestBid}
              bidUsers={auction.bidUsers}
              attributeType={auction.attributes.type}
            />
          ))
        )}
      </div>
      {deferredAuctions.length != 0 ? (
        <Button blocked={true} size="small">
          <div className={styles.loading}>
            <Icon name="loading" /> <span>Load more</span>
          </div>
        </Button>
      ) : null}
    </div>
  )
}
