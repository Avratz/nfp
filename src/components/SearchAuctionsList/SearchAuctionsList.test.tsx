import { render } from '@testing-library/react'
import { SearchAuctionsList } from './SearchAuctionsList'

jest.mock('@/ui/Icon/Icon', () => {
  function Icon() {
    return <div></div>
  }

  Icon.displayName = 'Icon'
  return {
    __esModule: true,
    Icon,
  }
})
describe('SearchAuctionsList', () => {
  it('renders SearchAuctionsList', () => {
    let { container } = render(<SearchAuctionsList />)
    expect(container).toMatchSnapshot()
  })
})
