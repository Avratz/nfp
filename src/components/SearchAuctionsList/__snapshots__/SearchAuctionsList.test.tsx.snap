// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAuctionsList renders SearchAuctionsList 1`] = `
<div>
  <div
    class="auctionCardsContainer"
  >
    <div
      class="auctionCards"
    >
      <div
        class="noMatch"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="160"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Ferror.png&w=640&q=75"
          srcset="/_next/image?url=%2Fimages%2Ferror.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Ferror.png&w=640&q=75 2x"
          style="color: transparent;"
          width="215"
        />
        <h3
          class="heading 2xl"
        >
          Im terrible sorry about all this.
        </h3>
        <h4
          class="heading base"
        >
          Please try again with another filter.
        </h4>
      </div>
    </div>
  </div>
</div>
`;
